computation_graph/run.py

```diff
@@ -204,16 +205,48 @@ def _get_computation_input(
     # For each edge, there are multiple values options, each having its own trace.
     values_for_edges_choice: Iterable[Iterable[_ChoiceOfOutputForNode]],
 ) -> base_types.ComputationInput:
-    bound_signature = _get_bound_signature(node.signature.is_args, incoming_edges)
+    incoming_edges_no_future = tuple(graph.remove_future_edges(incoming_edges))
+    bound_signature = _get_bound_signature(
+        node.signature.is_args, incoming_edges_no_future
+    )
     unbound_signature = _signature_difference(node.signature, bound_signature)
     results = gamla.pipe(
```

Comment:
i am thinking if we shouldn't do the changes in `values_for_edges_choice` 
this is where we compute the running options for a specific node, so why not incorporate future edges there as well. 
and then we don't need different logic for future vs non future arguments.
question is do we have a way to pass the state to that function ?
wdyt ?